modules/currency-l0/src/main/scala/org/tessellation/currency/l0/http/routes/DataBlockRoutes.scala

```diff
@@ -0,0 +1,38 @@
+package org.tessellation.currency.l0.http.routes
+
+import cats.effect.Async
+import cats.implicits.catsSyntaxEitherId
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+
+import org.tessellation.currency.DataUpdate
+import org.tessellation.currency.dataApplication.DataApplicationBlock
+import org.tessellation.currency.schema.currency.CurrencyBlock
+import org.tessellation.kernel._
+import org.tessellation.security.signature.Signed
+
+import io.circe.Decoder
+import org.http4s.HttpRoutes
+import org.http4s.dsl.Http4sDsl
+import org.http4s.server.Router
+
+final case class DataBlockRoutes[F[_]: Async](
+  mkCell: Either[Signed[CurrencyBlock], Signed[DataApplicationBlock]] => Cell[F, StackF, _, Either[CellError, Ω], _]
+)(implicit decoder: Decoder[DataUpdate])
+    extends Http4sDsl[F] {
+  import org.http4s.circe.CirceEntityCodec.circeEntityDecoder
+
+  private val httpRoutes: HttpRoutes[F] = HttpRoutes.of[F] {
+    case req @ POST -> Root / "l1-data-output" =>
+      req
+        .as[Signed[DataApplicationBlock]]
+        .map(_.asRight[Signed[CurrencyBlock]])
+        .map(mkCell)
+        .flatMap(_.run())
```

Comment:
Does `run` never fail? Maybe we should at least return BadRequest in case some error occurs.

Reply:
It would be good to have some kind of error returned here

Reply:
Good point. For DAG cell, we also do not check for errors so either it never fails of we missed something.

Reply:
I included basic error handling